packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -287,9 +306,11 @@ export class PipelineEditor extends React.Component<
   }
 
   updateModel(): void {
-    this.widgetContext.model.fromString(
-      JSON.stringify(this.canvasController.getPipelineFlow(), null, 2)
-    );
+    const pipelineFlow = this.canvasController.getPipelineFlow();
+
+    this.widgetContext.model.fromString(JSON.stringify(pipelineFlow, null, 2));
+
+    this.setState({ emptyPipeline: Utils.isNewPipeline(pipelineFlow) });
```

Comment:
does setState only update the given values or the whole state (ie will this remove `showPropertiesDialog` and `propertiesInfo` from the state)?

When testing it locally it doesn't seem to cause issues, but I'm still curious to the answer

Reply:
It updates the part of the state we want to update, so it will merge the object passed to setState with the current state, we don't lose the previous values